File: Hoops.Core/Models/Division.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace Hoops.Core.Models
{
    public partial class Division
    {
        [Key]
        public int DivisionId { get; set; }
        public Nullable<int> CompanyId { get; set; }
        public Nullable<int> SeasonId { get; set; }
        public string DivisionDescription { get; set; }
        public Nullable<int> DirectorId { get; set; }
        public Nullable<int> CoDirectorId { get; set; }
        public string Gender { get; set; }
        public Nullable<System.DateTime> MinDate { get; set; }
        public Nullable<System.DateTime> MaxDate { get; set; }
        public string Gender2 { get; set; }
        public Nullable<System.DateTime> MinDate2 { get; set; }
        public Nullable<System.DateTime> MaxDate2 { get; set; }
        public string DraftVenue { get; set; }
        public Nullable<System.DateTime> DraftDate { get; set; }
        public string DraftTime { get; set; }
        public bool Stats { get; set; }
        public Nullable<System.DateTime> CreatedDate { get; set; }
        public string CreatedUser { get; set; }

        // [ForeignKey("DirectorID")]
        // public virtual Person Director { get; set; }
    }
    
}

File: hoops.ui/src/domain/division.ts
export class Division {
  public seasonId!: number;
  public divisionId!: number;
  // public divisionName: string,
  public divisionDescription!: string;
  public minDate: Date | undefined;
  public maxDate!: Date | undefined;
  public gender: string = 'M';
  public minDate2!: Date | undefined;
  public maxDate2!: Date | undefined;
  public gender2: string = 'M';
  // Director is optional; use null when not set to avoid FK violations
  public directorId: number | null = null;
  public createdDate?: Date | string | null;
  public createdUser?: number | null;
  public modifiedDate?: Date | string | null;
  public modifiedUser?: number | null;
}
